### DISCUSSION TITLE
Manually upgrading upgrade requests to socket.io socket?

### DISCUSSION THREAD

**Discussion by Blakeinstein** [socketio/socket.io#4068] [Q&A] [answered]
I am trying to manually handle the socket upgrade request, so that I can proxy to another port on subdomains, and go to socket.io otherwise. Trying to use http2-proxy, but I cant find documentation on how to upgrade to socket.io socket
```ts
server.on('upgrade', (req, socket, head) => {
  if (req.headers.host ...)
    return proxy.ws(req, socket, head, { target ... });
  io.????? // upgrade to socket.io here
}
```
I also tried
```ts
io.use((socket, next) => {
  if (socket.request.headers.host ...)
    return ??? // Proxy to target port here
  next();
}
```
And I am confused as to what I should do to proxy socket.io socket

**Comment by darrachequesne** [comment, marked as answer]
You can manually upgrade with `io.engine.handleUpgrade()`.

Please see the documentation here: https://socket.io/docs/v4/server-api/#engine-handleUpgrade-request-socket-head

**Reply by Blakeinstein** [reply]
Thanks! BTW Why is `io.engine` of type `any`?

**Reply by Blakeinstein** [reply]
> 
> 
> > You can manually upgrade with `io.engine.handleUpgrade()`.
> > Please see the documentation here: [socket.io/docs/v4/server-api/#engine-handleUpgrade-request-socket-head](https://socket.io/docs/v4/server-api/#engine-handleUpgrade-request-socket-head)

I am unsure why you put in the effort of copying the solution post as is, and then replying back. Maybe I am missing some key information you wanted to add on here?
